src/zarr/storage/_obstore.py

```diff
@@ -196,6 +197,15 @@ async def delete(self, key: str) -> None:
         with contextlib.suppress(FileNotFoundError):
             await obs.delete_async(self.store, key)
 
+    async def delete_dir(self, prefix: str) -> None:
+        # docstring inherited
+        self._check_writable()
+        if prefix != "" and not prefix.endswith("/"):
+            prefix += "/"
+
+        keys = [(k,) async for k in self.list_prefix(prefix)]
```

Comment:
If you're not using the async iterator from the list, you could also just call `metas = await obs.list(self.store, prefix=prefix).collect_async()` and then extract the `path` from each dict. That might give a tiny bit less async overhead